### PR TITLE
feat(webhook): debug-log raw ADO body on allowlist rejection + edge-case tests for handler helpers

### PR DESCRIPTION
**Pull request by rios0rios0**
## Summary

Two changes in one commit:

1. **Diagnostic** — on `HandleAzureDevOps` allowlist rejection, emit a `Debug` log line with the parsed shape (`event_type`, `pull_id`, `status`, `repo_id`, `repo_name`, `remote_url`, `project_name`) plus a 4 KB head of the raw request body. Only fires on rejection; production at INFO stays quiet.
2. **Tests** — pin the contract of every unexported helper in the ADO handler with comprehensive edge-case coverage. New `export_test.go` re-exports them; new `azuredevops_internal_test.go` (in the external `webhooks_test` package) holds the matrix.

## Why

Live on `internal-terraform/internal-customer-app#NNNN`, deliveries get rejected with `org="" project=""` even though ADO's notification management API shows the same event with `repository.remoteUrl = "https://ExampleOrg@dev.azure.com/..."` and `repository.project.name = "internal-terraform"`. Local `json.Unmarshal` of the API-recorded event populates everything correctly. The actual HTTP body delivered to the pod must differ from what's in the API records — the diagnostic captures it so we can write a targeted parser fix.

The test additions cover every edge case we have actually hit at runtime, with a comment on each row explaining the production bug it pins. Future "let me clean this up" refactors surface in the test before they ship.

## Test plan

- [x] `go test -tags unit ./internal/infrastructure/controllers/webhooks/` — passes (33 new sub-cases + all existing)
- [x] `make lint` — 0 issues
- [ ] After deploy + a fresh push on `internal-customer-app#NNNN`, pod log emits the rejection diagnostic with the actual body
- [ ] Follow-up PR fixes the parser based on what the diagnostic reveals, with a regression test pinning the live-payload scenario

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/infrastructure/controllers/webhooks/azuredevops.go [hunk 1]
```diff
@@ -99,6 +99,23 @@ func (d *Dispatcher) HandleAzureDevOps(w http.ResponseWriter, r *http.Request) {
...
+			"event_type":    event.EventType,
+			"pull_id":       event.Resource.PullRequestID,
+			"status":        event.Resource.Status,
+			"repo_id":       event.Resource.Repository.ID,
+			"repo_name":     event.Resource.Repository.Name,
+			"remote_url":    event.Resource.Repository.RemoteURL,
+			"project_name":  event.Resource.Repository.Project.Name,
+			"body_length":   len(body),
+			"body_head_4kb": truncateForLog(string(body), adoRawBodyLogLimit),
+		}).Debug("ADO webhook: payload diagnostic on allowlist rejection")
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/infrastructure/controllers/webhooks/azuredevops.go:120]: Addressed in `a94116e`. The diagnostic now passes the raw `[]byte` directly into a new `support.TruncateBytesForLog(body, n)` helper, which converts only the first `min(n, len(body))` bytes to a string before quoting — a 50 MB hostile body now allocates ~4 KB for the log line, not a proportional intermediate string. New `support.TestTruncateBytesForLog/should_NOT_allocate_a_full-size_string_when_input_is_much_larger_than_the_cap` pins the budget.

[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/infrastructure/controllers/webhooks/azuredevops.go:120]: Addressed in `a94116e`. Both diagnostic raw-body logs (ADO handler + AI parser) now wrap the truncated content in `strconv.Quote` — newlines, ANSI escapes, CR-LF, and invalid UTF-8 are escaped before `logrus.TextFormatter` sees them, so a body containing `\nlevel=error msg="forged"` cannot inject a fake log line. `support.TestTruncateForLog/should_escape_newlines_so_they_cannot_inject_fake_log_lines` and the ANSI / CR-LF / invalid-UTF-8 cases pin the contract.

File: internal/infrastructure/controllers/webhooks/azuredevops.go [hunk 2]
```diff
@@ -211,3 +228,20 @@ func extractADOOrganization(remoteURL string) string {
...
+// input was clipped. Safe for arbitrary UTF-8 — the cut is byte-based, so a
+// trailing multi-byte rune may be split, which is acceptable for a raw-body
+// diagnostic where the goal is to glance at the structure, not parse it.
+func truncateForLog(s string, n int) string {
+	if len(s) <= n {
+		return s
+	}
+
+	return s[:n] + "...[truncated]"
+}
```
[the remark this answers is not in the document]
Comment (rios0rios0) [reply internal/infrastructure/controllers/webhooks/azuredevops.go]: Addressed in `a94116e`. Both helpers now live in a single `internal/support/truncate.go` file: `Truncate(s, n)` (plain), `TruncateForLog(s, n)` (quoted), `TruncateBytesForLog(b, n)` (byte-slice + quoted). Both `response_parser.go` and `azuredevops.go` consume from there; the local duplicates are gone, and the sentinel `...[truncated]` is defined as a single `truncationSentinel` const so a future rename has only one site to touch.
